[docs][data grid] Revise the Row Grouping doc (#16217)
mui/mui-x: docs/data/data-grid/tree-data/tree-data.md

## Group expansion

> Group expansion for tree data works the same as describe in [Row grouping—Group expansion](/x/react-data-grid/row-grouping/#group-expansion).

Author: ```suggestion
Group expansion for tree data works similarly to how it's described in [Row grouping—Group expansion](/x/react-data-grid/row-grouping/#group-expansion).
```
